### ISSUE TITLE
[SonicGenerations] 

### ISSUE THREAD

**Issue by sonicsonic5** [thesupersonic16/HedgeModManager#394]
HMM Info:
```
    Version: 7.7-0
    Args: 
    StartDir: C:\Program Files (x86)\SEGA\Sonic Generations\
    Process Level: User
    Game: Sonic Generations
    GameInstall: Sonic Generations
    Update Status: Up To Date
```

Exception:
```
    Type: UnauthorizedAccessException
    Message: Access to the path 'C:\Program Files (x86)\SEGA\Sonic Generations\profiles.json' is denied.
    Source: mscorlib
    Function: Void WinIOError(Int32, System.String)
    StackTrace: 
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
       at System.IO.StreamWriter.CreateFile(String path, Boolean append, Boolean checkHost)
       at System.IO.StreamWriter..ctor(String path, Boolean append, Encoding encoding, Int32 bufferSize, Boolean checkHost)
       at System.IO.File.InternalWriteAllText(String path, String contents, Encoding encoding, Boolean checkHost)
       at HedgeModManager.MainWindow.<SaveConfig>d__50.MoveNext() in C:\Users\SuperSonic16\source\repos\HedgeModManager\HedgeModManager\UI\MainWindow.xaml.cs:line 804
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
       at HedgeModManager.MainWindow.<UI_SaveAndPlay_Click>d__60.MoveNext() in C:\Users\SuperSonic16\source\repos\HedgeModManager\HedgeModManager\UI\MainWindow.xaml.cs:line 1048
    --- End of stack trace from previous location where exception was thrown ---
       at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
       at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
       at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
       at System.Windows.Threading.DispatcherOperation.InvokeImpl()
       at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Windows.Threading.DispatcherOperation.Invoke()
       at System.Windows.Threading.Dispatcher.ProcessQueue()
       at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
       at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
       at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
       at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
       at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
       at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
       at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
       at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
       at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
       at System.Windows.Application.RunDispatcher(Object ignore)
       at System.Windows.Application.RunInternal(Window window)
       at HedgeModManager.HedgeApp.Main(String[] args) in C:\Users\SuperSonic16\source\repos\HedgeModManager\HedgeModManager\HedgeApp.xaml.cs:line 352
    InnerException: 
```

**Comment by sonicsonic5** [comment]
gAAAAJxTGEikR2R7+a+xiOxsc+GS20mt4mnR9Dr1FL0J8iNRTB15wsUTp3PLRjV0IYOxHqe3U0/yU5Dl0WsukjDjr8SNZKFk6sKNBwQQ1eQ+pG6lUCwPIuW3hrPfF62s3vP4/3FzZriPefv1EQ/WJ4YSKcIZ2cBSMKNUxYQthMuKge2VgAAAAKoTJZrj9bUimGUJ6WDBDF4f2/u/sydwRuBrj3kz7QUvKP4F2qZuul+zD24PyD/YsbW8LR5tWoCFfXLeuYTepEryiX2nNEI5PnkfD/Klx5ceF9peNguN8Dy8EUlv9aIPIH94OlfbVMt3Gc27K+IMb9eQWaiN2hkEb9fomQFYAbpRRTRMxzgKS3VM5T/13uPyIF9GqJQUJohECkIH84b9jIlt0a1Zx+EUEQXUmdnLoLfBDd3Q+N138QN8uOva7NL5OHDGXIs8Ask6qME0BkOeDM6KSxBJJ5v7ECTIaRuiEwYvQJ3fCd/JHmx/TTRH9eLZiGiJY3iomE18ZAx5PCYAqHc3pQ6DG5YfHMSVkBaHalR0Jk7oew9xmJZr0DIf3Hi6SwSsM/JLV+OZoVLX/sMeHEG+ZgCF6Ibr/X2eJH5kiijhftX8aiFg1ycCQ+WxzRD8+4MRhIb152B3pTLk+4kt9xdBoda2BnoohSvppgJcZ2g0kCZ2fYGlP5S1RUfPdrcpcz5NzTW2f0FV4k70EaQA0q2qM0UcE3rax7vMeDE+70KoMhCiIvvyLvT6B9zlfbb0+W7LnJhfAzPHxCs9UViFDFrHJzKLL4ZxGxNrgL4raGMCwFh1EeVUQVgRbRZmjBmTld9nQN/qzq8kMr0nbMEPe1cyaS0QO1DFOhrVF9kNbjcwfpnv4ZOvrPmR3TiDwLdECh8SbBnHn118nGeZlwMJSlioACJwKoUrMwqeOa4mW7/i2+iMqcNh8ofOn7+W8CMioRS3p2g6Asdd6aS/AFYp0oOTmRM56Zl/wIypkaP+7O1QUPS7AvuXD8M2tmnARlpW4Zw3FDgrGLk0NfRdsKsih2AyreEeYoSqhHVzgTJ3bSSmksjIn9HAE6DjVx8RXPIkPvTkFac9oFAXZgUbQHxXUuZ94dFuzvyhl6lGYoXCfjTblxia2KqRN4wyB/qlrq2xTgU4JPF5ABUYvOtL9/Pd1PaenWrbplpdQzcGGol5nzNsjYm4ho16XrqK/O9OA/m8lwBGkqQdacdG7c6w8aIcWeEza6k034/A5EK8kQ4EEBFP7x30ucoC682F11apAzbK7ir0iOljOcjLplNCVMpLr3HPwko6gqwVMd1a6o6JU8IbSPEPkh8wZgEBaFyI08fGvQrnMaR4BWxT8mkkHuChNR7U643ECBE/7eGLBfPC5dZYgxx7e2uQWydf6J7NM+DBhPoiLYLdrd1DFhPrwnibjLjfU4Yjq6P11XbpLwhkdn3XDwtstie9ZEa/B7Etqnfj6qO9tgSw7XZWbgJWnS8JEtuFUo3HdOzWho+CbLKkgHyQ053oQIv8zuPLJPzBRFSNsiPygXQTVnOCt10V2D+kLq8TOdlj4DVWV/9+ydGKpXxw7WYE4uSevGG5qoQvrR2Z2XDBGwofOcfC0jRs3cGLiq7M5BWfzKgmA07YXS2UcogVUKIVPieagdcN1C01yC6qyzaQaGyFiJBL/F7Rc9J2lTPJOYZejzJt8Ii/KKPn3ryq43BQm1jz9OFiJpwKW+Vh8MJitZglFIVCJhrsZebCbC8AWayrgY00CxSM7m6jxloaSrVVB9iWUFBko79KbUjMt5zQ6OcR5HQmUA4Dakpz0AcXmJifLnpd5dQZ31sjqzvDSuysa8ldBL+ZOHyOzuTBi5AYdEMnAdAzn03GbSnTMPXceqWKX/yXoCMNt5tHqgb3/QVx+QRvBYGLRi60TdZTdpFZPdAep+d8rXZs75iHoJ6IRv6hYwzwEsJEcS5P0OcpLhIJbvYx7eRDWFpWqCyqU5Un8IqXYcKprvbI88a4nCoG5mLrHGaesdeEDRMMh+BJHtKLVIqo7b2gMZADhGmuVM4J3VyN5pMJ1FWSEf16PzYPD3qpn1GDIKmdObspx32oXKzK3vnwiGTnNUKxmfO30uVkz3+p1HV4z7fodSBLquSHp2jprhRTbWU/HMMzdVYLTC2zrbURUm5pTZuziU7ccJQJ2QpHolVX3io46umrPxGkDQQYmPS2tp98E5DAz1pmGlLDQXpbzCvtIA3C3uV7HJH4yCnAZIJLqn2ru2za6vwGykC4Nz910YcFXGawZNjWWE+usKistxymaQkcVieW195aEYk0+u99yo6l4FEfLNkPBipLixcrLx2dSm06pDjFIeRTsBtNs3dKUOWAmj4E9U9izymNs7F7rsx+b6eQZZ9uoQo5YuYHfNqpqU0KWRu9M9Ua/nz+HX3MF+bFLlHhkFqYqcKp4teanfz4iNHqnDZUsHxpnCfNOA90ydITTaP18S3dfC19ps2uveP1ehgdkCUyvzGuc53EkZk9Kjw1z3ExS+1/Vk9Xy3SGTLhnIUWnK/vRWrO4tPSeL5Oo6HRavoSN1QcGnDHevR8keYQMWump6oVOAtsmQge64cq0io9n3Pg7I/RZuAgcCwVt91jMpJ/MttUGVBSbuzHHZVxWN1SPBghtlroXaUTZ4o1yZE2TDDXDL/Wb2AgAuEmK2cxSKRMGduIAs5rdHL9rNN6RZ+X0qrCi1L19eIRVmqKz6pt8L34YLksZJnp9AqOIKsx+/GN7QAKZ8TI3ZSu8fNpSniFrmTepw38OOcIoX2tbvKNtnOXWKEhRIn4GlPHe1yt/VmV0i3K4mR1bw/1q1NufzEMnZNLuGXFAgfLrWtwHYOHprWfOEp9QXc27g+SdZO+8RPX6+Q4J8Wg5vZQpEwxZW+ECP0SdileaT8MJ/6lvtmHBIM/+Fkipd+GubpkO3EP5b9UCKroiL5/YUP6RMXj9S7+95oQt7PMzY1ckso0Lq7bA2QYHGjPbF22pkD7Ffk0detymfCgVc5CJSvRf888iTcXJDpbUDb9sm0E28oZnd6+W8PU/3WIHRiMfmwnJ2vT8VokVKl8W9BfbKRzcW6tSPAc7G3hikuUWItQofb/5vJTHlStNoM0Oxtsoc+SA/rV/ga/sl6JeGmQvnnbTjTeloZcbagmjBveUrm9HJRXBXF7+u5nPWfloRMQiF/wsdEzA5uNr3w2o4G9KdzlIg89G1zHKVYnyMz4qSGh7u/HIVqK0kXW3qFc3/dmdaWeQ6RiTqe0Cuv0izpPnfjwJP0/QjicmhGOIy9z9TPullf3J//ruV8pZB3B7FEj/clv8yKI8UBs/1ZRW6HhfJS8T9EbYhJmL5Ua0KLKrydwnv6kHYzdDhqwiiYdpzjeqa+eO7LmGQ4Ujgwr/HUXAXR2tTZ609bOSBnoZhkp9ErK4jyUoKeOTNVPgnmoUiWihT1glgn56cTEKsatflClTDvxVbvmFpE0t2USPpnliD5n66p8HWssAC12gGBrm7XExfc4gMMhtA5eKWJP5mxdFWr/CLUHdRBAON+jBsmutEBmmqxCguVL2XSy+kENqlFCHAU8TJBl95y4bw/mlhnMjeHNu9rO/Tk3HxrpJENiaBbW8Nmgavsgm1xRnbHc8U+hDxW9z0dgx8xGDJ7hz0WwRYB6mJ0wNdYpKHcHbzz4v8ktI2bNe/nd2rY/ljXe4s1O5L51JV0ckXgmxwsLx8XZn0wYxions/Er+08HGGeaUESlWl8n1S2yOdd03UmLSkGiOV6rJ+sRVf4oVd/lNyCK/4lLAK6la+poh7/NbgEK+8lAdLBWUQk62/lIqn1wxQbumLa95WVxMyxr7YmhKqxOCQxfoOt2AiPT3DSFhQsEAhrNCEvfv31/g2SLmXM14K3tr+BUy4VZC+BxNhcPV9qancs/usX1ehXh67TPMeQ+1rVFxfPasXBPxvoOkAjgyJqQBIcYo4CrfJ2WRfQp9F8Ws1MbXqwo8VJUzwGCWsNHmKPSwXIrgJEV9UGpDO+UUqApGMe0cWUYyTeoum42iSgyQ/SFWYW/7x5W09XzlAbJKC5GOgh0QKW2BsRHDtqJngfkEsgxeDPr0VfG/g+KFdYWJRd0BukUde7ggHYLOGje/SGTq8OKOuBgG9nTZjQOH+neGoChNSTx/6czQ8tL/gbo5AU+7T8ooA67dThyNRrTUAsoSoZdOLro5Mq0h7aFTzHwRHQbXhRwEHpWzvrXG5s9NGOaGkAn4oa/x+ymvIbFAaeJJmOkQypxmpAy+tt2HzZt/girxk0jxLg==

**Comment by thesupersonic16** [comment]
We do not support non-Steam/pirated versions of Sonic Generations. Please install the game normally using Steam.

This crash has been fixed in e3e2f8013e130cdc4b981d289639110a260d9594 which has not yet been released.

Closing.
